Client/src/Remote/Endpoints/DatasetEndpoint.tsx

```diff
@@ -49,6 +49,25 @@ export const approvedDataset = async (query: IFlaggedDatasetQuery) => {
     SnackbarUtils.success(`Dataset ${query.datasetId} was approved!`)
   }
 }
+
+export const callGetUserFavouriteDatasets = async (): Promise<number[]> => {
+  return get(userSavedDatasetsRoute).json()
+}
+
+export const userSaveFavouriteDataset = async (datasetId: number) => {
+  const result = await post(userSavedDatasetsRoute + '/' + datasetId).json()
+  if (result == 'Favorite data set successfully saved') {
+    SnackbarUtils.success(`Dataset ${datasetId} was saved in favourites!`)
```

Comment:
remove the ${datasetId} please, since the users shouldnt be aware what the id is of the dataset